src/Mvc/Mvc.Analyzers/src/TopLevelParameterNameAnalyzer.cs

```diff
@@ -122,6 +128,25 @@ internal static bool IsProblematicParameter(in SymbolCache symbolCache, IParamet
             return false;
         }
 
+        private static bool IsComplexType(in SymbolCache symbolCache, ITypeSymbol type)
+        {
+            // This analyzer should not apply to simple types. In MVC, a simple type is any type that has a type converter that returns true for TypeConverter.CanConvertFrom(typeof(string)).
+            // Unfortunately there isn't a Roslyn way of determining if a TypeConverter exists for a given symbol or if the converter allows string conversions.
+            // https://github.com/dotnet/corefx/blob/v3.0.0-preview8.19405.3/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L103-L141
+            // provides a list of types that have built-in converters. Of the types that can convert from string, everything with the exception of three reference types is a value type.
+            // We'll use a simpler heuristic in the analyzer: A type is simple if it's a value type or if it's one of the the following types: System.CultureInfo, System.Uri, and System.Version.
+            // Using value types as parameters to be bound is fairly rare, so it's fairly limited 
```

Comment:
Yeah, may want to remove the value type special-case because non-`System` value types may not have associated type converters and the message already mentions the cases where the analyzer can be ignored.